scipy/optimize/_optimize.py

```diff
@@ -259,6 +259,56 @@ def __dir__(self):
         return list(self.keys())
 
 
+class OptimizeResult(_OptimizeResult):
+    def warn_deprecation_jac(self):
+        message = ('Use of attribute/item `jac` is deprecated and replaced '
+                   'by `grad`.  Support for `jac` will be removed in SciPy '
+                   '1.13.0.')
+        warnings.warn(DeprecationWarning(message))
```

Comment:
Think `stacklevel=3` is correct here because this method is called elsewhere within the result but it would be worth checking
```suggestion
        warnings.warn(DeprecationWarning(message), stacklevel=3)
```

Reply:
Thanks. I don't know how I always forget to include a stacklevel after this long...